src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -675,6 +679,23 @@ void BaseCouplingScheme::doImplicitStep()
     // no convergence achieved for the coupling iteration within the current time window
     if (_acceleration) {
       _acceleration->performAcceleration(getAccelerationData());
+      /**
+       * Acceleration changes CouplingData::values(), so we must override the data in CouplingData::_timeStepsStorage.
+       * There are generally two possibilities:
+       *
+       * 1) Only override the data in CouplingData::_timeStepsStorage that is part of the receive data
+       * 2) Override the data in CouplingData::_timeStepsStorage for all data in getAccelerationData()
+       *
+       * We are using strategy 2), because it's easier to access getAccelerationData() from here. However, this also means that
+       * we have to make sure that the send data is stored in the CouplingData::_timeStepsStorage - even though this is not
+       * needed at the moment. Important note: In https://github.com/precice/precice/pull/1414 also send data requires to keep
+       * track of _timeStepsStorage for subcycling. So it will become simpler as soon as subcycling is fully implemented.
+       */
+      // @todo For other Acceleration schemes as described in "Rüth, B, Uekermann, B, Mehl, M, Birken, P, Monge, A, Bungartz, H-J. Quasi-Newton waveform iteration for partitioned surface-coupled multiphysics applications. Int J Numer Methods Eng. 2021; 122: 5236– 5257. https://doi.org/10.1002/nme.6443" we need a more elaborate implementation.
+      for (auto &pair : getAccelerationData()) {
+        bool override = true;
+        pair.second->storeDataAtTime(pair.second->values(), time::Storage::WINDOW_END, override);
+      }
```

Comment:
The reason for the last failing test was here: After acceleration we again have to store the accelerated data in the storage, because the acceleration only updates `Mesh::Data`, not `CouplingData::_timeStepsStorage`. This also required some updates of how the storage of send data is managed. See 5f789ac952af9f2aa39ec3da7cfd24a71b2325e4 for the complete picture. I think it should be possible to clean this up a bit more, but most important for now: It's working as expected.

This issue was only a problem for parallel coupling schemes, because only in parallel coupling schemes it is allowed to accelerate send data. For receive data this problem was not visible, because the data in `Mesh::Data` was updated correctly and this is also the data that finally arrives at the user.